File: Python/E965_UnivaluedBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
# 56 ms
class Solution:
    def postOrder(self, root, val, res):
        if root:
            self.postOrder(root.left, val, res)
            self.postOrder(root.right,val, res)
            res[0] &= (root.val == val)
    def isUnivalTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        res = [True]
        self.postOrder(root, root.val, res)
        return res[0]
        

File: Cpp/E965.UnivaluedBinaryTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    void postOrder(TreeNode* root, int val, bool &res){
        if(root){
            postOrder(root->left,val,res);
            postOrder(root->right,val,res);
            res &= (root->val == val);
        }
    }
    bool isUnivalTree(TreeNode* root) {
        bool res = true;
        postOrder(root,root->val,res);
        return res;
    }
};